katdal/concatdata.py

```diff
@@ -444,10 +442,31 @@ def __setitem__(self, name, data):
             for n, cache in enumerate(self.caches):
                 cache[name] = data[self._segments[n]:self._segments[n + 1]]
 
-    def iterkeys(self):
+    def __delitem__(self, name):
+        found = False
+        for cache in self.caches:
+            try:
+                del cache[name]
+                found = True
+            except KeyError:
+                pass
+        if not found:
+            raise KeyError
```

Comment:
Do you need to pass `name` to `KeyError` to match the error message from `dict`?

Reply:
Good idea - done.